hibernate-core/src/test/java/org/hibernate/jpa/test/query/QueryTest.java

```diff
@@ -1451,4 +1483,41 @@ public void testGetSingleResultWithNoResultException() {
 			entityManager.close();
 		}
 	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-11409")
+	public void testParameterRegisterredCollection() {
+		final EntityManager em  = getOrCreateEntityManager();
+		try {
+			em.getTransaction().begin();
+			{
+				Item item = new Item( "LongList", "[5,11,6123,-61235,24]" );
+				em.persist( item );
+				// deallocate item after braces close
+			}
+			em.getTransaction().commit();
+
+			em.getTransaction().begin();
+			LongList ll = new LongList();
+			ll.add( 5L );
+			ll.add( 11L );
+			ll.add( 6123L );
+			ll.add( -61235L );
+			ll.add( 24L );
+			Query tq = em.createNativeQuery( "SELECT * FROM Item WHERE descr = ?", Item.class );
+			tq.setParameter(1, ll);
```

Comment:
This is wrong. The Java type is `String` so the parameter should be set as:

    .setParameter(1, "[5,11,6123,-61235,24]")

in which case it works just fine even after I revert your changes to `AbstractProducedQuery`.

Your test should use the `LongList` as an entity attribute for the `TypeContributor` to have a purpose in this test, right?